tests/connection_state_test.c

```diff
@@ -564,56 +780,109 @@ static int s_test_mqtt_connection_timeout_fn(struct aws_allocator *allocator, vo
 
     ASSERT_INT_EQUALS(AWS_ERROR_MQTT_TIMEOUT, state_test_data->interruption_error);
 
-    struct aws_array_list *received_messages =
```

Comment:
These diffs are a mess.
I converted the validatation part of mqtt_connection_timeout to use the helper. And added a new test `mqtt_connection_any_publish `